frontend/src/features/seeds/routes/CreateSeed.tsx

```diff
@@ -2,20 +2,48 @@ import PageLayout from '../../../components/Layout/PageLayout';
 import CreateSeedForm from '../components/CreateSeedForm';
 import useCreateSeedStore from '../store/CreateSeedStore';
 import { NewSeedDto } from '@/bindings/definitions';
+import { SelectOption } from '@/components/Form/SelectMenu';
 import PageTitle from '@/components/Header/PageTitle';
 import SimpleModal from '@/components/Modals/SimpleModal';
-import { useState } from 'react';
+import usePreventNavigation from '@/hooks/usePreventNavigation';
+import { useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
+function useDebounce<T>(searchFunction: (searchParam: T) => void, delay: number) {
+  const [timeoutId, setTimeoutId] = useState<ReturnType<typeof setTimeout>>();
+
+  const debouncedSearch = (searchParam: T) => {
+    if (timeoutId) {
+      clearTimeout(timeoutId);
+    }
+
+    const timeout = setTimeout(() => {
+      searchFunction(searchParam);
+    }, delay);
+
+    setTimeoutId(timeout);
+  };
+
+  return [debouncedSearch];
+}
```

Comment:
Debouncing might be useful for other components, maybe put this in the global `utils` folder

Reply:
I agree.
I also changed the hook, so that it can be easily integrated into react-query, once we merge this.